### PR TITLE
fix(curriculum): Attribute order shouldn't matter Music Player

### LINKED ISSUE #53130: Attribute order shouldn't matter Music Player Step 72
### Describe the Issue

It doesn't seem right to force campers to make `onclick` the first attribute. In perfectly fine HTML and JavaScript, the attributes can be placed in any order.  

### Affected Page

https://www.freecodecamp.org/learn/javascript-algorithms-and-data-structures-v8/learn-basic-string-and-array-methods-by-building-a-music-player/step-72

### Your code

```
     <button class="playlist-song-delete" aria-label="Delete ${song.title}" onclick="${deleteSong(song.id)}>
          <svg width="20" height="20" viewBox="0 0 16 16" fill="none" xmlns="http://www.w3.org/2000/svg"><circle cx="8" cy="8" r="8" fill="#4d4d62"/>
          <path fill-rule="evenodd" clip-rule="evenodd" d="M5.32587 5.18571C5.7107 4.90301 6.28333 4.94814 6.60485 5.28651L8 6.75478L9.39515 5.28651C9.71667 4.94814 10.2893 4.90301 10.6741 5.18571C11.059 5.4684 11.1103 5.97188 10.7888 6.31026L9.1832 7.99999L10.7888 9.68974C11.1103 10.0281 11.059 10.5316 10.6741 10.8143C10.2893 11.097 9.71667 11.0519 9.39515 10.7135L8 9.24521L6.60485 10.7135C6.28333 11.0519 5.7107 11.097 5.32587 10.8143C4.94102 10.5316 4.88969 10.0281 5.21121 9.68974L6.8168 7.99999L5.21122 6.31026C4.8897 5.97188 4.94102 5.4684 5.32587 5.18571Z" fill="white"/></svg>
        </button>
```

### Expected behavior

This code should pass the challenge as it is completely valid. The curriculum helpers should be able to locate any attribute on an embedded HTML element inside a JavaScript function, or regex should be able to recognize that the value was set. 

### Screenshots

_No response_

### System

- Device: Desktop
- OS: Windows 10
- Browser: Firefox
- Version: 121.0.1

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by DivyaMaddipudi**
Attribute order shouldn't matter Music Player Step 72 #53130

Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #53130

### CODE REVIEW

**Review by JoyShaheb — CHANGES_REQUESTED** [review]

File: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-basic-string-and-array-methods-by-building-a-music-player/6557421eb6a7a0f0500e3106.md [hunk 1]
```diff
@@ -7,14 +7,14 @@ dashedName: step-72
...
+Within the button element in the `renderSongs` function, add an `onclick` attribute. For the value, call the `deleteSong` function and interpolate `song.id`.
 
 # --hints--
 
-You should add an `onclick` attribute as the first attribute of the delete button and pass in `deleteSong(${song.id})`.
+You should add an `onclick` attribute as an attribute of the delete button and pass in `deleteSong(${song.id})`.
 
 ```js
-assert.match(code, /<button\s+onclick=('|")deleteSong\(\$\{song\.id\}\)\1\s*class=('|")playlist-song-delete\2\s*aria-label=('|")Delete\s*\$\{song\.title\}\3\s*>/)
+assert.match(code, /<button(?=.*\bclass="playlist-song-delete")(?=.*\baria-label="Delete [^"]*")(?=.*\bonclick="deleteSong\([^)]*\)")\s+(?:class|aria-label|onclick)="[^"]*"\s+(?:class|aria-label|onclick)="[^"]*"\s+(?:class|aria-label|onclick)="[^"]*"\s*>/)
```
[the remark this answers is not in the document]
Comment (DivyaMaddipudi) [reply curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-basic-string-and-array-methods-by-building-a-music-player/6557421eb6a7a0f0500e3106.md:17]: @JoyShaheb, I tried your suggestion but it is failing though onclick is present as an attribute anywhere

[the remark this answers is not in the document]
Comment (JoyShaheb) [reply curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-basic-string-and-array-methods-by-building-a-music-player/6557421eb6a7a0f0500e3106.md:17]: can you try this ?
```suggestion
assert.match(code, /<button(?=(?:(?=.*\bclass=['"]playlist-song-delete['"])(?=.*\baria-label=['"]Delete [^'"]*['"])(?=.*\bonclick=['"]deleteSong\([^)]*\)['"]))(?:.*\s+(?:class|aria-label|onclick)=['"][^'"]*['"])\s*>)/);
```

[the remark this answers is not in the document]
Comment (DivyaMaddipudi) [reply curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-basic-string-and-array-methods-by-building-a-music-player/6557421eb6a7a0f0500e3106.md]: This works! Updated the code.

**Review by JoyShaheb — APPROVED** [review]
well done ! @gikf and @jdwilkin4  can you guys have a final look ?

**Review by gikf — COMMENTED** [review]
I cannot say I'm not confused how this works, but it indeed appears to allow the various ordering of the attributes.

Regex though doesn't check if the `song.title` and `song.id` are used as interpolated values - anything can be used in their place.

**Review by moT01 — CHANGES_REQUESTED** [review]
I think we need to check what @gikf mentioned here @DivyaMaddipudi - e.g. the code should not pass with `onclick="deleteSong()"`.

**Comment by jdwilkin4** [comment]
Hey @DivyaMaddipudi ! 

It looks like a couple of reviews were left for this PR

Did you have any questions about what changes needed to be made?

**Comment by Ksound22** [comment]
Thanks for your contribution, @DivyaMaddipudi.

We've moved on from this cuz another PR has fixed the issue.

**Pull request closed without merging**